.github/workflows/release.yml

```diff
@@ -0,0 +1,59 @@
+---
+name: Build and Optionally Release
+
+on:  # yamllint disable-line rule:truthy
+  pull_request:
+  push:
+    branches:
+    - main
+
+jobs:
+  release:
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Set up QEMU
+      id: qemu
+      uses: docker/setup-qemu-action@v2
+      with:
+        image: tonistiigi/binfmt:latest
+        platforms: all
+    - name: Setup buildx
+      uses: docker/setup-buildx-action@v2
+    - name: Build Only
+      uses: docker/bake-action@v2
+      with:
+        files: |
+          ./docker-bake.hcl
+        targets: build
+    - name: Release
+      id: release
+      uses: ahmadnassri/action-semantic-release@v2
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Docker meta
+      id: meta
+      uses: crazy-max/ghaction-docker-meta@v4
+      with:
+        images: ${{ github.repository }}
+        tags: |
+          type=semver,pattern={{version}},value=${{ steps.release.outputs.release-version }}
+          type=ref,event=branch
+          type=sha
+    - name: Login to DockerHub
+      if: steps['release']['outputs']['published'] == 'true'
```

Comment:
Just an FYI, you can also do `steps.release.outputs.published` in addition to the array indexing.